modules/dnn/test/test_int.cpp

```diff
@@ -6,9 +6,54 @@
 // Third party copyrights are property of their respective owners.
 
 #include "test_precomp.hpp"
+#include <opencv2/dnn/shape_utils.hpp>
 
 namespace opencv_test { namespace {
 
+template <class T>
+Mat randomIntMatImpl(const std::vector<int> &sizes, int64_t low, int64_t high, int matType)
+{
+    CV_Assert(low < high);
+    CV_Assert(high - low < RAND_MAX);
+
+    Mat m(sizes, matType);
+
+    int size = 1;
+    for (int s: sizes) {
+        size *= s;
+    }
+
+    T* data = m.ptr<T>();
+    for (int i = 0; i < size; ++i)
+    {
+        data[i] = low + std::rand() % (high - low);
+    }
+
+    return m;
+}
+
+Mat randomIntMat(const std::vector<int> &sizes, int64_t low, int64_t high, int matType)
+{
+    if (matType == CV_32S)
+        return randomIntMatImpl<int32_t>(sizes, low, high, matType);
+    else if (matType == CV_64S)
+        return randomIntMatImpl<int64_t>(sizes, low, high, matType);
+    else
+        CV_Error(Error::BadDepth, "Unsupported type");
+    return Mat();
+}
+
+int64_t getValueAt(const Mat &m, const int *indices)
```

Comment:
I think this method is not very generic

Reply:
It will be used only for integer tests in this file.
test_int.cpp will contain a lot of simple test with int operations.
Each test should have int32 and int64 versions. Each test will use randomIntMat to generate random values with int32 or int64 type with appropriate range and will use gatValueAt to check values of int32 or int64 type.